File: LarkCardKit/Builders/OverflowBuilder.cs
using LarkCardKit.Models.Behaviors;
using LarkCardKit.Models.Elements;

namespace LarkCardKit.Builders;

public class OverflowBuilder
{
    private readonly Overflow _overflow = new();
    
    public OverflowBuilder AddOption(string value, string text)
    {
        _overflow.Options ??= new List<OverflowOption>();
        _overflow.Options.Add(new OverflowOption
        {
            Value = value,
            Text = new PlainText { Content = text }
        });
        return this;
    }
    
    public OverflowBuilder Width(string width)
    {
        _overflow.Width = width;
        return this;
    }
    
    public OverflowBuilder Disabled(bool disabled = true)
    {
        _overflow.Disabled = disabled;
        return this;
    }
    
    public OverflowBuilder ElementId(string id)
    {
        _overflow.ElementId = id;
        return this;
    }
    
    public OverflowBuilder Margin(string margin)
    {
        _overflow.Margin = margin;
        return this;
    }
    
    public OverflowBuilder OnClick(object callbackData)
    {
        _overflow.Behaviors ??= new List<object>();
        _overflow.Behaviors.Add(new CallbackBehavior { Value = callbackData });
        return this;
    }
    
    public Overflow Build() => _overflow;
}

File: TypeScript/LarkCardKit/src/builders/OverflowBuilder.ts
import type { Overflow, OverflowOption } from '../models/elements';

export class OverflowBuilder {
  private _overflow: Overflow = { tag: 'overflow', options: [] };

  option(text: string, value: string): this {
    this._overflow.options ??= [];
    this._overflow.options.push({
      text: { tag: 'plain_text', content: text },
      value
    });
    return this;
  }

  options(options: Array<{ text: string; value: string }>): this {
    this._overflow.options = options.map(opt => ({
      text: { tag: 'plain_text' as const, content: opt.text },
      value: opt.value
    }));
    return this;
  }

  elementId(id: string): this {
    this._overflow.element_id = id;
    return this;
  }

  margin(margin: string): this {
    this._overflow.margin = margin;
    return this;
  }

  build(): Overflow {
    return this._overflow;
  }
}
